modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -204,7 +204,13 @@ def __init__(
         self._dtypes = dtypes
 
         self._validate_axes_lengths()
-        self._filter_empties(compute_metadata=False)
+        if all(obj is not None for obj in (index, columns, row_lengths, column_widths)):
+            # this hint allows to filter empty partitions out without triggering metadata computation
+            # in order to avoid useless computation over empty partitions
+            compute_metadata = True
+        else:
+            compute_metadata = False
```

Comment:
I'm not getting it, doesn't this condition inside `._filter_empties` means the same?
https://github.com/modin-project/modin/blob/6afee33df4871578e0056dde614878b4870f8d2a/modin/core/dataframe/pandas/dataframe/dataframe.py#L471-L478

I mean, it won't trigger metadata computation anyway if `compute_metadata=False` was passed, does it?

Reply:
The conditions are different. Here we want to filter empty partitions out if all the metadata is available (compute_metadata=True) so we won't trigger computation.

Reply:
I still don't understand what's happening here.

Your condition says:
1. If all metadata is available pass `compute_metadata=True`.
2. If there's not enough metadata to filter empties pass `compute_metadata=False`.

The condition inside `.filter_empties` says:
1. If all metadata is available filter empty partitions.
2. If `compute_metadata=False` and there's not enough metadata to filter empties do nothing.
3. If `compute_metadata=True` trigger metadata computation anyway in order to filter empties.

I don't understand why these new conditions are needed. Shouldn't it perform identically the same if you would just keep passing `compute_metadata=False` all the time?

Reply:
correct, removed